test/ConnectionTest3.cpp

```diff
@@ -0,0 +1,77 @@
+#include <gtest/gtest.h>
+
+#include "DBConnection3.hpp"
+#include "DBConnection3Caster.hpp"
+
+namespace {
+
+    int getMySqlConnection3Info(IDBConnection3& dbConnection) {
+        auto& mysql_connection = DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::cast(dbConnection);
+        return mysql_connection.query();
+    }
+
+    std::string getMySqlConnection3AdvancedInfo(IDBConnection3& dbConnection) {
+        auto& mysql_connection = DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::cast(dbConnection);
+        return mysql_connection.advancedQuery().server_version;
+    }
+
+    int getSqLiteConnection3Info(IDBConnection3& dbConnection) {
+        auto& sqlite_connection = DBConnection3Caster<SqLiteDBConnection3, IDBConnection3>::cast(dbConnection);
+        return sqlite_connection.query();
+    }
+
+    void checkMySqlConnection3Info(IDBConnection3& dbConnection, CastType& assertion, int& expectedResult) {
```

Comment:
@Morheit ,
-  `CastType&` -> pass by value. this is just enum
-  rename assertion -> expected_cast_type
- int& expectedResult -> pass by value. this is just in.
- rename expectedResult -> expected_value

Reply:
- IDBConnection3& can it be const?

Reply:
@malirod , done. No. Constness of  `IDBConnection3&` will require changes in `DBConnectionCaster3`